nemo_skills/evaluation/metrics/code_metrics.py

```diff
@@ -29,31 +35,24 @@ def update(self, predictions):
                 The content of the file is benchmark specific.
         """
         self.total += 1
-
-        if len(predictions) > 1:
-            self.agg_mode = f"pass@{len(predictions)}"
-
-            self.total_correct += any([elem['is_correct'] for elem in predictions])
-            self.total_correct_plus += any([elem['is_correct-plus'] for elem in predictions])
-        else:
-            # If single prediction, set it to greedy aggregation mode
-            self.agg_mode = "greedy"
-
-            self.total_correct += predictions[0]['is_correct']
-            self.total_correct_plus += predictions[0]['is_correct-plus']
+        self.get_pass_at_k(self.agg_mode_dict, predictions=predictions)
 
     def get_metrics(self):
-        metrics_dict = {
-            "num_entries": self.total,
-            "passing_base_tests": self.total_correct / self.total * 100.0,
-            "passing_plus_tests": self.total_correct_plus / self.total * 100.0,
-        }
+        metrics_dict = {}
+        for agg_mode, agg_metric_dict in self.agg_mode_dict.items():
```

Comment:
seems that every class has this loop now, maybe let's move that over to base and then each metric will only need to define a new method that takes agg_metric_dict and returns the metric dictionary that's then assigned to metrics_dict[agg_mode] by the base class? But current way is also ok, I guess, not too much duplication

Also I feel that our variable names here could be improved, so feel free to change self.agg_mode_dict / agg_metric_dict to something more descriptive if you have better ideas